code/modules/reagents/reagent_containers/spray.dm

```diff
@@ -108,6 +107,21 @@
 	desc = "BLAM!-brand non-foaming space cleaner!"
 	list_reagents = list("cleaner" = 250)
 
+/obj/item/reagent_containers/spray/cleaner/safety
+	desc = "BLAM!-brand non-foaming space cleaner! This spray bottle can only accept space cleaner."
+
+/obj/item/reagent_containers/spray/cleaner/safety/on_reagent_change()
+	var/found_filthy_reagent = FALSE
+	for(var/datum/reagent/R in reagents.reagent_list)
+		if(R.id != "cleaner") //all chems other than space cleaner are filthy.
+			reagents.del_reagent(R.id)
+			found_filthy_reagent = TRUE
+		if(found_filthy_reagent)
```

Comment:
The use of the `found_filthy_reagent` var is not really needed here. For example, if the `if(R.id != "cleaner")` check passes, you know it found a filthy reagent, there's no need to set the var to true and then check if its true.